### PR TITLE
Add fast path to podFitsOnNode

### PR DESCRIPTION
**Pull request by mrkm4ntr**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug

/kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:
`podFitsOnNode` can exit 2nd iteration immediately if `addNominatedPods` changes no data.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: pkg/scheduler/core/generic_scheduler.go [hunk 1]
```diff
@@ -590,15 +590,17 @@ func addNominatedPods(pod *v1.Pod, meta predicates.PredicateMetadata,
...
 		if util.GetPodPriority(p) >= util.GetPodPriority(pod) && p.UID != pod.UID {
 			nodeInfoOut.AddPod(p)
 			if metaOut != nil {
 				metaOut.AddPod(p, nodeInfoOut)
 			}
+			podsAdded = true
 		}
 	}
-	return true, metaOut, nodeInfoOut
+	return podsAdded, metaOut, nodeInfoOut
```
Comment (wgliang) [inline pkg/scheduler/core/generic_scheduler.go]: Can you add any unit tests for this change?

**Review by wgliang — CHANGES_REQUESTED** [review]

File: pkg/scheduler/core/generic_scheduler.go (hunk 1 above)
Comment (mrkm4ntr) [reply pkg/scheduler/core/generic_scheduler.go]: This change is for performance improvement. In k8s, is such change a target of unit tests?

Comment (mrkm4ntr) [reply pkg/scheduler/core/generic_scheduler.go]: @wgliang I added unit tests for this change. Review please.

**Comment by wgliang** [comment]
/lgtm
Thanks @mrkm4ntr

**Review by bsalamat — COMMENTED** [review]
/lgtm
/approve

Thanks, @mrkm4ntr!

**Comment by mrkm4ntr** [comment]
Thanks @bsalamat! But why the approved label was removed?

**Comment by bsalamat** [comment]
> But why the approved label was removed?

Oh, that must be mistake/UI confusion. I added the label again.

**Pull request merged**